gufe/mapping.py

```diff
@@ -1,19 +0,0 @@
-# This code is part of OpenFE and is licensed under the MIT license.
-# For details, see https://github.com/OpenFreeEnergy/gufe
-
-from openff.toolkit.utils.serialization import Serializable
-
-
-class Mapping(Serializable):
-    """A mapping, usually of atoms, between two `ChemicalSystem`s.
-
-    Some, but not all, transformation protocols require a mapping to execute.
-
-    """
```

Comment:
So here @dotsdl was thinking about mappings as between Systems (collection of components) rather than just two Components.  Worth discussing.

Reply:
Yeah so in the context of gufe, I agree with this. Atom mappings may span multiple components (i.e. you _should_ have the ability to perturb a side chain whilst also perturbing a ligand, or in the more reasonable case, perturb an ion to account for a ligand with a net charge changing at the same time).

Having the atom mapping be system wide could be easier to control the "link" in the mappings between the various components? Alternatively you could just bake in a reference to the component being perturbed in the mapping object and then you'd feed in a frozenset of mapping objects or something like that?

Reply:
I have a stupid question/point regarding the first comment by @richardjgowers. A non particle related mapping could be something like a thermodynamic change  (temperature or permitivity) or a conformational description (alpha-helix  or pi-helix).  Is that something you were searching for?

Reply:
possibly but thermodynamic control directly, but one may possibly want to have some means of encoding some collective variable of interest maybe (including possibly a titration experiment in a CpH,EMD style calculation). That being said I don't think this really should be under "mapping". I'm not sure we've really started fleshing out things beyond alchemical control but those two cases probably should go under a separate CV class.

One type of thing we could conceivably think as going under a "mapping" are restraints. One could think of applying bond/angle/dihedral restraints as a type of mapping.

Reply:
shamless plug for an restraint finding algorithm and a "empty mapping case": https://github.com/rinikerlab/restraintmaker